src/ui/utils/sanitize.ts

```diff
@@ -43,12 +43,22 @@ export function sanitize(obj: any, path: string, category: string, logSanitized:
   return obj;
 }
 
+export function sanitizeState(state: any, logSanitized: boolean) {
+  return sanitize({ ...state, ast: undefined }, "", "state", logSanitized);
+}
+
+export function sanitizeAction(action: any, logSanitized: boolean) {
+  return action.type === "SET_SYMBOLS"
+    ? { type: action.type }
+    : sanitize(action, "", `action[${action.type}]`, logSanitized);
+}
+
 export const sanityCheckMiddleware: Middleware<
   {},
   UIState,
   Dispatch<UIAction>
 > = store => next => action => {
-  sanitize(store.getState(), "", "state", true);
-  sanitize(action, "", `action[type=${action.type}]`, true);
+  sanitizeState(store.getState(), true);
```

Comment:
I'm worried about sanitizing the state for every action. This feels like redundant work i.e. we just checked the a source in the sources store in the last round, we shouldn't need to do it again.

This is also unnecessary because the fronts would be arriving in the store via an action and we just checked them on the way in.

Reply:
The `sanitize()` function was primarily written for the prod case where we need to sanitize state and actions before they are processed by LogRocket. Note that all the fronts will remain in the redux state, only the copies of actions and state that we pass to LogRocket are modified.

Reply:
got it - good catch.

Perhaps we minimize the LogRocket middleware because we're not using it too much at the moment:

```js
actionSanitizer: action => ({ type: action.type }
stateSanitizer: state => () => ({})
```

This should help us with perf and then we can slowly add data in that would be useful

Reply:
I'd like to take one more look at this using Karina's recording but if it keeps impacting our performance, only handing parts of the state and actions to LogRocket sounds like a good idea.